### DISCUSSION TITLE
Using referer header for enhanced security

### DISCUSSION THREAD

**Discussion by GaryAustin1** [supabase/supabase#8613] [Feature Requests]
Many uses of Supabase are from a specific website.  

Currently it looks like one can use PostgREST http settings in database calls like `(current_setting('request.headers', true)::json->>'referer')::text` to enforce access only from certain websites.  

My quick tests show this comes thru to RLS and rpc functions.  The referer header by my understanding can not be changed by javascript code so should be secure.  I've used it in a PHP API a while back to limit access.

So... 
First is this a valid way to further limit access to the database for anon access?
And second (for the team) how hard is this to add to gotrue and storage as a setting for trigger functions and RLS to examine?

In the end, the ability to set an overall list of referers that can access the entire infrastructure seems to be ideal.

Am I missing something?

**Comment by liaujianjie** [comment]
Are you trying to implement something like CORS?

**Reply by GaryAustin1** [reply]
Well if Supabase had an option to apply a domain to cors, that would solve it also.  I believe right now they return '*' so any domain can access.

**Reply by steve-chavez** [reply]
>  I believe right now they return '*' so any domain can access.

Yes, related to https://github.com/PostgREST/postgrest/issues/2441

**Reply by GaryAustin1** [reply]
@steve-chavez Not sure how your comment impacts the Supabase hosted platforms.  But even if it was something that can be set for PostgREST (which saves RLS dealing with it with the header check... so good!).  It would be a big win, IMO, that any requests (auth in particular) could be limited to specific domains.  Maybe the 3rd party redirect stuff is the issue with auth?.....

I just assume I'm missing something, as ability to set a limit on access to a specific domain seems basic.

Or maybe the dashboard/studio is an issue?

**Reply by steve-chavez** [reply]
Hm, so for Supabase itself it might be better to set CORS at the proxy level(cloudflare worker) for all the services? Though as you mention perhaps auth needs a different configuration.

@J0 Would it be hard to add CORS as a config for GoTrue?

**Reply by GaryAustin1** [reply]
Also sort of wondering about all the different ways the API's get called within Supabase infrastructure itself.  I think storage goes thru PostgREST for the object table.  Not sure on 3rd party auth flow, if there are non-mysite originated calls.   Deno functions, if they call the database.  Database http extension calls say to storage.  

Maybe be a tough list to maintain at the top level.

**Comment by leohxj** [comment]
does firebase support CORS settings?
